lib/reek/report/report.rb

```diff
@@ -121,6 +123,41 @@ def total_smell_count_message
     end
 
     #
+    # Displays the status of each file as it is examined
+    # Displays the text report when complete
+    #
+    # @public
+    class ProgressReport < TextReport
```

Comment:
Do you feel that modelling the option (see above) as separate Report class?
If yes, we can go with this way as well. I just think that it's important that we show it as an option to the user to not confuse him.